pkg/cni/cilium/cilium.go

```diff
@@ -0,0 +1,285 @@
+		},
+		Rules: []rbac.PolicyRule{
+			{
+				APIGroups: []string{"networking.k8s.io"},
+				Resources: []string{"networkpolicies"},
+				Verbs:     []string{"get", "list", "watch"},
+			},
+			{
+				APIGroups: []string{""},
+				Resources: []string{"namespaces", "services", "nodes", "endpoints", "componentstatuses"},
+				Verbs:     []string{"get", "list", "watch"},
+			},
+			{
+				APIGroups: []string{""},
+				Resources: []string{"pods", "nodes"},
+				Verbs:     []string{"get", "list", "watch", "update"},
+			},
+			{
+				APIGroups: []string{"extensions"},
+				Resources: []string{"networkpolicies", "thirdpartyresources", "ingresses"},
+				Verbs:     []string{"get", "list", "watch", "create"},
+			},
+			{
+				APIGroups: []string{"apiextensions.k8s.io"},
+				Resources: []string{"customresourcedefinitions"},
+				Verbs:     []string{"create", "get", "list", "watch", "update"},
+			},
+			{
+				APIGroups: []string{"cilium.io"},
+				Resources: []string{"ciliumnetworkpolicies", "ciliumnetworkpolicies/status", "ciliumendpoints", "ciliumendpoints/status"},
+				Verbs:     []string{"*"},
+			},
+		},
+	}
+
+	clusterRoleBinding = rbac.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: CiliumClusterRoleName,
+		},
+		RoleRef: rbac.RoleRef{
+			APIGroup: rbac.GroupName,
+			Kind:     "ClusterRole",
+			Name:     CiliumClusterRoleName,
+		},
+		Subjects: []rbac.Subject{
+			{
+				Kind:      rbac.ServiceAccountKind,
+				Name:      CiliumServiceAccountName,
+				Namespace: metav1.NamespaceSystem,
+			},
+			{
+				Kind: "Group",
+				Name: "system:nodes",
+			},
+		},
+	}
+
+	clusterRoleBindingPSP = rbac.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: CiliumClusterRoleNamePSP,
+		},
+		RoleRef: rbac.RoleRef{
+			APIGroup: rbac.GroupName,
+			Kind:     "ClusterRole",
+			Name:     "suse:kubic:psp:privileged",
+		},
+		Subjects: []rbac.Subject{
+			{
+				Kind:      rbac.ServiceAccountKind,
+				Name:      CiliumServiceAccountName,
+				Namespace: metav1.NamespaceSystem,
+			},
+		},
+	}
+
+	ciliumBaseName   = "cilium-etcd-client"
+	ciliumCertConfig = certutil.Config{
+		CommonName: "cilium-etcd-client",
+		Usages:     []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth},
+	}
+)
+
+func init() {
+	// self-register in the CNI plugins registry
+	cni.Registry.Register("cilium", EnsureCiliumAddon)
+}
+
+// EnsureCiliumAddon creates the cilium addons
+func EnsureCiliumAddon(cfg *config.KubicInitConfiguration, client clientset.Interface) error {
+	etcdDir := strings.Join([]string{cfg.Certificates.Directory, "/etcd"}, "")
+	var ciliumConfigMapBytes, ciliumDaemonSetBytes []byte
+
+	caCert, caKey, err := pkiutil.TryLoadCertAndKeyFromDisk(etcdDir, "ca")
+	if err != nil {
+		fmt.Printf("etcd generation retrieval failed %v", err)
+	}
+
+	cert, key, err := pkiutil.NewCertAndKey(caCert, caKey, &ciliumCertConfig)
+	if err != nil {
+		fmt.Printf("error when creating etcd client certificate for cilium %v", err)
```

Comment:
Same here: abort if we cannot create a new certificate and key.

```suggestion
		return fmt.Errorf("error when creating etcd client certificate for cilium %v", err)
```